packages/components/pie-thumbnail/src/index.ts

```diff
@@ -27,21 +29,46 @@ export class PieThumbnail extends LitElement implements ThumbnailProps {
     @property({ type: String })
     public alt = defaultProps.alt;
 
+    @property({ type: Boolean })
+    public disabled = defaultProps.disabled;
+
+    @property({ type: Boolean })
+    public hasPadding = defaultProps.hasPadding;
+
+    @property({ type: String })
+    @validPropertyValues(componentSelector, backgroundColors, defaultProps.backgroundColor)
+    public backgroundColor = defaultProps.backgroundColor;
+
+    @property({ type: Object })
+    public placeholder: ThumbnailProps['placeholder'];
+
+    private _handleImageError () {
+        if (this.placeholder?.src) this.setAttribute('src', this.placeholder.src);
+        if (this.placeholder?.alt) this.setAttribute('alt', this.placeholder.alt);
+    }
+
     render () {
         const {
             variant,
             src,
             alt,
+            disabled,
+            hasPadding,
+            backgroundColor,
+            _handleImageError,
         } = this;
 
         const wrapperClasses = {
             'c-thumbnail': true,
             [`c-thumbnail--${variant}`]: true,
+            [`c-thumbnail-background--${backgroundColor}`]: true,
```

Comment:
quick thing - just create a map here of the prop value to the camelCase class name. Then everything works as Ash has asked and its hardly any extra work/code as far as I can tell

Reply:
i will do that in a follow up PR shortly